frontend/src/pages/user/UserAnswerPage.js

```diff
@@ -86,7 +88,9 @@ export default function UserAnswerPage() {
         console.log(error);
       }
     };
-    calculate();
+    if (currentPage === lastPage) {
+      calculate();
+    }
```

Comment:
I dont know why you used other condition when you already hove isDone variable for that

```suggestion
    if (isDone) {
      calculate();
    }
```